### ISSUE TITLE
[SC2086] False positive with variables containing the number of arguments

### ISSUE THREAD

**Issue by orbea** [koalaman/shellcheck#1362]
#### For bugs
- Rule Id: `SC2086`
- My shellcheck version: `online`
- [x] I read the issue's wiki page, e.g. https://github.com/koalaman/shellcheck/wiki/SC2086
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

set -- 1 2 3
arg=$#

echo $arg

```

#### Here's what shellcheck currently says:

```
Line 6:
echo $arg
     ^-- SC2086: Double quote to prevent globbing and word splitting.
```

#### Here's what I wanted or expected to see:

Nothing.

My understanding is that `$arg` contains the number of arguments (`$#`) and will always be a number that will not suffer globbing or word splitting. Is this correct?

**Comment by koalaman** [comment]
Well caught. `$#` is indeed always a number, and will be counted as such for SC2086 from now on.
